### PR TITLE
feat(ci/codecov): after_n_builds per flag & codecov comment in PRs

### PR DESCRIPTION
**Pull request by armenzg**
Codecov now supports `after_n_builds` per flag rather than a shared value.

This means that FE & BE changes would not create a codecov check on PRs until the exact number of codecov checks have uploaded their results. FE changes currently upload 5 codecov results while BE changes upload 11.

We will have to be careful to update the number of builds for each as more CI checks are added.

This [BE PR](https://github.com/getsentry/sentry/pull/46943) (based on this branch) created a status check:
image

There are some issues with [the FE PR](https://github.com/getsentry/sentry/pull/46944) that I will figure out in a follow up PR.

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -1,10 +1,3 @@
-codecov:
-  notify:
-    # The minimum amount of codecov reporting checks that need to complete
-    # For FE we have 5 checks and for BE we have 11
-    # TODO: Remove this when we have support of after_n_builds per flag
-    after_n_builds: 5
```
Comment (armenzg) [inline codecov.yml:6]: There will now be two different values in the flags section.

File: codecov.yml [hunk 2]
```diff
@@ -15,16 +8,14 @@ coverage:
       frontend:
         # codecov will not fail status checks for master
         only_pulls: true
-        # This will be needed until codecov supports after_n_builds for flags
-        informational: true # Do not fail the check
+        informational: false # Fail the check
```
Comment (armenzg) [inline codecov.yml:11]: The codecov CI checks will be back and will fail if they do not hit a 60% FE coverage.

File: codecov.yml [hunk 3]
```diff
@@ -59,9 +50,28 @@ flags:
...
+    # Do not send any status checks until N coverage reports are uploaded
+    after_n_builds: 11
 
-comment: false
+# after_n_builds declared in flags cannot control when to make comments
+# A value of 11 will mean that only BE PRs will get a comment (only if the score changes)
+# Once comments listen to the value in flags we can add comments to FE PRs
+# Read more here: https://docs.codecov.com/docs/pull-request-comments
+comment:
+  after_n_builds: 11
```
Comment (armenzg) [inline codecov.yml:67]: Only BE PRs will have a comment *only* if there is coverage changes.

File: codecov.yml [hunk 4]
```diff
@@ -59,9 +50,28 @@ flags:
...
+  layout: "diff, files"
+  # Update, if comment exists. Otherwise post new.
+  behavior: default
+  # require_changes: true
```
Comment (mattgauntseo-sentry) [inline codecov.yml:71]: Not this line can be removed - its uncommented below

File: codecov.yml [hunk 5]
```diff
@@ -15,16 +8,14 @@ coverage:
...
+        informational: false # Fail the check
         target: 60%
         flags:
         - frontend
       backend:
         # codecov will not fail status checks for master
         only_pulls: true
-        # This will be needed until codecov supports after_n_builds for flags
-        informational: true # Do not fail the check
+        informational: false # Fail the check
```
Comment (mattgauntseo-sentry) [inline codecov.yml:18]: There would be value in breaking this change into two parts:

1. Enable the comments
2. Set informational to false

That was if there are any issues with the coverage status check, reverting it won't disable the comments.

File: codecov.yml [hunk 6]
```diff
@@ -59,9 +50,28 @@ flags:
     paths:
     - "static/app/"
     carryforward: true
+    # Do not send any status checks until N coverage reports are uploaded
```
Comment (mattgauntseo-sentry) [inline codecov.yml:53]:
```suggestion
    # FE uploads 5 coverage reports. This property ensures codecov waits
    #  for all reports to be uploaded before creating a GitHub status check.
```

**Review by mattgauntseo-sentry — APPROVED** [review]

**Pull request merged**